validator/src/signer.rs

```diff
@@ -0,0 +1,91 @@
+//! Signing-related interface and the sample software-only implementation.
+
+use ed25519_consensus::SigningKey;
+use rand_core::{CryptoRng, RngCore};
+use tendermint::{account, PrivateKey, PublicKey, Signature};
+
+use crate::error::Error;
+
+/// The trait for different signing backends
+/// (HSMs, TEEs, software-only, remote services etc.)
+#[tonic::async_trait]
+pub trait SignerProvider {
+    type E: std::error::Error;
+    async fn sign(&self, signable_bytes: &[u8]) -> Result<Signature, Self::E>;
+    async fn load_pubkey(&self) -> Result<PublicKey, Self::E>;
+}
```

Comment:
FWIW this is close to the trait offered by [`async_signature::Signer`](https://docs.rs/async-signature/latest/async_signature/trait.AsyncSigner.html).